### DISCUSSION TITLE
Is JSX/TSX supported?

### DISCUSSION THREAD

**Discussion by alexgorbatchev** [ast-grep/ast-grep#307] [General]
I can't figure out if JSX/TSX patterns are supported. It's obvious that grepping through .tsx files works, but I can't figure out how to create a JSX patterns, for example `<div>$$$</div>`. Should that work?

**Comment by HerringtonDarkholme** [comment]
Of course they are supported. See the playground.

https://ast-grep.github.io/playground.html#eyJtb2RlIjoiUGF0Y2giLCJsYW5nIjoiamF2YXNjcmlwdCIsInF1ZXJ5IjoiPGRpdj4kJCQ8L2Rpdj4iLCJjb25maWciOiIjIENvbmZpZ3VyZSBSdWxlIGluIFlBTUxcbnJ1bGU6XG4gIGFueTpcbiAgICAtIHBhdHRlcm46IGlmIChmYWxzZSkgeyAkJCQgfVxuICAgIC0gcGF0dGVybjogaWYgKHRydWUpIHsgJCQkIH1cbmNvbnN0cmFpbnRzOlxuICAjIE1FVEFfVkFSOiBwYXR0ZXJuIiwic291cmNlIjoiZnVuY3Rpb24gQ29tcG9uZW50KCkge1xuICByZXR1cm4gPGRpdj5IZWxsbyBXb3JsZDwvZGl2PlxufSJ9

**Reply by alexgorbatchev** [reply]
Thanks for the example! Switching to TypeScript however doesn't appear to be working. Is that expected?

**Reply by HerringtonDarkholme** [reply]
The playground does not include the tsx grammar. The language list is the same as https://tree-sitter.github.io/tree-sitter/playground.

**Reply by HerringtonDarkholme** [reply]
You can use ast-grep command line to test tsx. Say `sg -p '<div>$$$</div>' -l tsx`

**Reply by alexgorbatchev** [reply]
Thanks! For some reason I can't get this to work in my codebase. Will keep trying.

**Reply by HerringtonDarkholme** [reply]
Now the playground support tsx! https://github.com/ast-grep/ast-grep.github.io/pull/135 
(No syntax highlight though)
